pkg/controller/actuator.go

```diff
@@ -305,58 +291,6 @@ func (a *actuator) Migrate(ctx context.Context, log logr.Logger, ex *extensionsv
 	return a.Delete(ctx, log, ex)
 }
 
-func (a *actuator) reconcileVPNEnvoyFilter(
-	ctx context.Context,
-	alwaysAllowedCIDRs []string,
-	istioNamespace string,
-	istioLabels map[string]string,
-) error {
-	aclMappings, istioLabelsFromExt, err := a.getAllShootsWithACLExtension(ctx, istioNamespace)
-	if err != nil {
-		return err
-	}
-
-	// build EnvoyFilter object as map[string]interface{}
-	// because the actual EnvoyFilter struct is a pain to type
-	name := "acl-vpn"
```

Comment:
It seems we need to delete the legacy `EnvoyFilter` as it is not automatically cleaned up by a `ManagedResource`. How is this accomplished?

Reply:
Ok, this sounds reasonable. 

> In my tests, the shoot-specific part is removed during the second reconciliation because the resource manager needs some time to create the shoot-specific envoyfilter.

This is even what we want in an ideal implementation. As the new filter is created asynchronously, we should only remove the old filter part once the new one is in place. Otherwise, we might have shoots where the ACL is not active for a bit. I know this is only the VPN endpoint, but it still sounds like an elegant solution.

There shouldn't be any problems with checking the ACL twice in both the legacy and the new filter, correct?